feat: Add default event columns to EventDefinition model

HamedMP opened the pull request:
Add support for specifying default columns to display for an event in the data table. This includes:
- Added default_columns field to the EnterpriseEventDefinition model
- Created a migration for the new field
- Updated serializers to include default columns in API responses
- Implemented column saving logic in the column configurator
- Distinguished between global default columns and event-specific default columns
- Updated UI to use event-specific default columns when available


https://github.com/user-attachments/assets/48b0283e-009e-42ec-88bb-35c4f88e0d71

## Problem

<!-- Who are we building for, what are their needs, why is this important? -->

<!-- Does this fix an issue? Uncomment the line below with the issue ID to automatically close it when merged -->
<!-- Closes #ISSUE_ID -->

Re #25494 

At the moment there is no place to save the column configuration for the event table. (we only have `live_events_columns` in the Team level)

## Changes

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

<!-- Yes / no / it doesn't have an impact. -->

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

HamedMP committed:
feat(event-definitions): Add default columns support for event definitions

- Extend EnterpriseEventDefinition model with default_columns field
- Update serializer to include default_columns in representation
- Add default_columns to frontend event definition type
- Implement column saving logic for event-specific default columns in column configurator
- Update DefinitionView to use event-specific default columns when available

HamedMP committed:
update default column

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

HamedMP committed:
update the text + comply with the checkbox to save as default

HamedMP committed:
improve column saving, distinguish between where the column config is used

HamedMP committed:
remove old model definition from event_definition

HamedMP commented on frontend/src/queries/nodes/DataTable/ColumnConfigurator/columnConfiguratorLogic.tsx:
```diff
@@ -63,10 +66,38 @@ export const columnConfiguratorLogic = kea<columnConfiguratorLogicType>([
-            if (props.isPersistent && values.saveAsDefault) {
+        save: async () => {
+            // Check if we're in an event definition view
+            const isEventDefinition = router.values.currentLocation?.pathname.includes('/data-management/events/')
```
i really don't like this, curious how else we can check where the logic is being called, if not the router

HamedMP committed:
Revert "remove old model definition from event_definition"

This reverts commit 6578913a0e8ec62568a181db9cbfeeec335af077.

HamedMP committed:
fix(definition-view): Correct string interpolation syntax in definition description

HamedMP commented on frontend/__snapshots__/scenes-app-insights--funnel-top-to-bottom-edit--dark.png:
why the values in this funnel are not loaded anymore? i don't think this can be related to the current branch 👀

HamedMP commented on frontend/__snapshots__/scenes-app-insights--funnel-top-to-bottom-edit--dark.png:
why the values in this funnel are not loaded anymore? i don't think this can be related to the current branch 👀

aspicer commented on frontend/src/scenes/data-management/definition/DefinitionView.tsx:
```diff
@@ -32,25 +33,42 @@ export const scene: SceneExport = {
 }
 
+// Type guard to check if a Definition has default_columns with values
+function hasDefaultColumns(definition: Definition): definition is EventDefinition & { default_columns: string[] } {
```
This seems a little too defensive in its coding style to me. We've defined the types of Definition and control the stack - just checking if it's set or not should be good enough without this elaborate of a type guard.

aspicer commented on frontend/src/scenes/data-management/definition/DefinitionView.tsx:
```diff
@@ -232,7 +250,12 @@ export function DefinitionView(props: DefinitionLogicProps = {}): JSX.Element {
-                    <Query query={memoizedQuery} />
+                    <Query
+                        query={memoizedQuery}
+                        key={`event-definition-query-${definition.id}-${definition.updated_at}-${JSON.stringify(
```
What's the goal of this key field here? Is there an issue with the query not updating when the columns change?

aspicer commented on frontend/src/scenes/data-management/definition/definitionLogic.ts:
```diff
@@ -144,4 +144,13 @@ export const definitionLogic = kea<definitionLogicType>([
     }),
+
+    // Ensure we reload when navigating between definitions
+    propsChanged(({ actions, values, props }, oldProps) => {
```
Similar question here - is there an issue with navigating between events currently where it doesn't update properly? Or when does this get triggered and was it an issue before?

aspicer commented:
This looks like a good start! Left some questions about some things that I was confused by or thought should maybe be a little different, but generally looks great!

HamedMP committed:
feat(column-configurator): Refactor column configuration context and saving logic

- Add context prop to distinguish between event definition and team column configurations
- Remove router dependency in column configurator logic
- Update migration to use TextField for default_columns
- Simplify column saving logic in columnConfiguratorLogic
- Improve type safety and context handling in DefinitionView

aspicer commented on frontend/src/scenes/data-management/definition/DefinitionView.tsx:
```diff
@@ -32,25 +33,37 @@ export const scene: SceneExport = {
-    const memoizedQuery = useMemo(
-        () => ({
+    const memoizedQuery = useMemo(() => {
+        const columnsToUse = hasDefaultColumns(definition)
```
a function feels heavy here.

`const hasDefaultColumns = definition?.default_columns?.length` works?

aspicer commented:
FYI at least I'm getting some build errors:

```
@posthog/frontend:start: [TYPEGEN] 💔 frontend/src/queries/nodes/DataTable/dataTableLogic.ts:191:17 - error TS2322: Type '{ full: boolean; expandable: boolean; embedded: boolean; propertiesViaUrl: boolean; showPropertyFilter: boolean | TaxonomicFilterGroupType[]; showEventFilter: boolean; showSearch: boolean; showActions: boolean; showDateRange: boolean; showTestAccountFilters: boolean; showExport: boolean; showReload: boolean; showTimings: boolean; showElapsedTime: boolean; showColumnConfigurator: boolean; showPersistentColumnConfigurator: boolean; showSavedQueries: boolean; showHogQLEditor: boolean; allowSorting: boolean; showOpenEditorButton: any; showResultsTable: boolean; hiddenColumns: string[]; response?: NonNullable<Record<string, any> | EventsQueryResponse | ActorsQueryResponse | HogQLQueryResponse | WebOverviewQueryResponse | WebStatsTableQueryResponse | WebExternalClicksTableQueryResponse | WebGoalsQueryResponse | WebVitalsPathBreakdownQueryResponse | SessionAttributionExplorerQueryResponse | ErrorTrackingQueryResponse | ExperimentFunnelsQueryResponse | ExperimentTrendsQueryResponse | TracesQueryResponse | undefined> | undefined; context?: { type: "event_definition" | "team_columns"; eventDefinitionId?: string | undefined; } | undefined; kind: NodeKind.DataTableNode; columns: string[]; source: EventsNode | PersonsNode | EventsQuery | ActorsQuery | HogQLQuery | WebOverviewQuery | WebStatsTableQuery | WebExternalClicksTableQuery | WebGoalsQuery | WebVitalsPathBreakdownQuery | SessionAttributionExplorerQuery | ErrorTrackingQuery | ExperimentFunnelsQuery | ExperimentTrendsQuery | TracesQuery | WebVitalsQuery<InsightQueryNode>; }' is not assignable to type 'Required<Omit<DataTableNode, "response">>'.
@posthog/frontend:start: [TYPEGEN]   Types of property 'context' are incompatible.
@posthog/frontend:start: [TYPEGEN]     Type '{ type: "event_definition" | "team_columns"; eventDefinitionId?: string | undefined; } | undefined' is not assignable to type '{ type: "event_definition" | "team_columns"; eventDefinitionId?: string | undefined; }'.
@posthog/frontend:start: [TYPEGEN]       Type 'undefined' is not assignable to type '{ type: "event_definition" | "team_columns"; eventDefinitionId?: string | undefined; }'.
```

aspicer commented on frontend/src/queries/nodes/DataTable/ColumnConfigurator/columnConfiguratorLogic.tsx:
```diff
@@ -56,17 +62,44 @@ export const columnConfiguratorLogic = kea<columnConfiguratorLogicType>([
                 },
             },
         ],
+        context: [
```
I think you want this to be a selector and not a reducer. See `annotationsOverlayProps` in `annotationsOverlayLogic` as an example

HamedMP committed:
rebase to master

HamedMP committed:
add context to return

HamedMP committed:
remove function

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

HamedMP committed:
move context to selector

HamedMP committed:
update schemas for frontend and backend

aspicer commented:
Please consider adding the guard so that the query doesn't error if you're not on ee!

HamedMP commented:
@aspicer  absolutely, i'll fix it tomorrow 🙌🏼

HamedMP committed:
Merge branch 'master' into hamed-25494-fix-event-list-columns

HamedMP commented:
i believe it's much easier to add the guard after merging to master, i need to rebase a lot of files to get the features from billing that's merged to master. i'll add it directly after merging

aspicer commented:
@HamedMP in the future when you have that issue, just merge origin/master into your feature branch instead of merging your feature branch into master and then creating a new branch. We squash and merge the commit history anyway, so it's effectively going to the same as a rebase.

HamedMP commented:
true that, i realized it in the very end, just merged master into mine 🙌🏼 
<img width="881" alt="image" src="https://github.com/user-attachments/assets/5482f4da-e513-4e97-8b9b-e0be4783bd40" />

aspicer commented:
I don't think that is what I was saying.  You said:
> i believe it's much easier to add the guard after merging to master, i need to rebase a lot of files to get the features from billing that's merged to master. i'll add it directly after merging

I was telling you that you don't have to merge this branch into master to get to that start with the new billing changes, you can just merge master into your branch and add the guard?

HamedMP commented:
okay gotta, i'll do that in these situations, definitely much easier than rebasing (also editor-performance friendlier 😄)
